python/paddle/distributed/passes/pipeline_scheduler_pass.py

```diff
@@ -64,56 +81,324 @@ def partial_programs(self, program):
+            splitted_backward_programs,
+        ) = self._split_program_for_overlapping(
+            BACKWARD, backward_program, backward_split_points
+        )
+        (
+            splitted_forward_job_types,
+            splitted_forward_programs,
+        ) = self._split_program_for_overlapping(
+            FORWARD, forward_program, forward_split_points
+        )
+
+        self._multistreaming_for_overlapping(
+            splitted_backward_programs, BACKWARD
+        )
+        self._multistreaming_for_overlapping(splitted_forward_programs, FORWARD)
+
+        # Rearrange splitted chunks for BACKWARD and FORWARD
+        self.jobs_in_stable_phase.clear()
+        num_splitted_backward_jobs, num_splitted_forward_jobs = len(
+            splitted_backward_job_types
+        ), len(splitted_forward_job_types)
+        for idx in range(
+            max(num_splitted_backward_jobs, num_splitted_forward_jobs)
+        ):
+            if idx < num_splitted_backward_jobs:
+                self.jobs_in_stable_phase.append(
+                    splitted_backward_job_types[idx]
+                )
+            if idx < num_splitted_forward_jobs:
+                self.jobs_in_stable_phase.append(
+                    splitted_forward_job_types[idx]
+                )
+
+        return (
+            splitted_backward_job_types,
+            splitted_backward_programs,
+            splitted_forward_job_types,
+            splitted_forward_programs,
+        )
+
+    def _create_job_list(self):
         num_micro_batches = self.get_attr("num_micro_batches")
         pp_stage = self.get_attr("pp_stage")
         pp_degree = self.get_attr("pp_degree")
 
         job_list = []
-        lr_job = core.Job("lr")
+        lr_job = core.Job(LR)
         job_list.append(lr_job)
 
         assert (
             pp_degree <= num_micro_batches
-        ), "Num of micro batches should larger than pp degree."
+        ), "Num of micro batches should larger than or equal to pp degree."
 
         micro_batch_in_warmup = pp_degree - pp_stage
         micro_batch_in_1f1b = num_micro_batches - micro_batch_in_warmup
 
         forward_micro_batch_id = 0
         for i in range(micro_batch_in_warmup):
-            forward_job = core.Job("forward")
+            forward_job = core.Job(FORWARD)
             forward_job.set_micro_batch_id(forward_micro_batch_id)
             job_list.append(forward_job)
             forward_micro_batch_id += 1
 
         backward_micro_batch_id = 0
         for i in range(micro_batch_in_1f1b):
-            backward_job = core.Job("backward")
-            backward_job.set_micro_batch_id(backward_micro_batch_id)
-            job_list.append(backward_job)
-            backward_micro_batch_id += 1
-            forward_job = core.Job("forward")
-            forward_job.set_micro_batch_id(forward_micro_batch_id)
-            job_list.append(forward_job)
+            for job_type in self.jobs_in_stable_phase:
```

Comment:
1F1B-Overlap-Pass and 1F1B-Pass can be decoupled, cause their schedules are different.